wkb.go

```diff
@@ -0,0 +1,285 @@
+	if !ok {
+		return ErrUnsupportedDataType
+	}
+
+	if len(data) == 21 {
+		// the length of a point type in WKB
+		return p.unmarshalWKB(data)
+	}
+
+	if len(data) == 25 {
+		// Most likely MySQL's SRID+WKB format.
+		// However, could be a line string or multipoint with only one point.
+		// But those would be invalid for parsing a point.
+		return p.unmarshalWKB(data[4:])
+	}
+
+	return ErrIncorrectGeometry
+}
+
+func (p *Point) unmarshalWKB(data []byte) error {
+	if len(data) != 21 {
+		return ErrNotWKB
+	}
+
+	littleEndian, typeCode, err := scanPrefix(data)
+	if err != nil {
+		return err
+	}
+
+	if typeCode != 1 {
+		return ErrIncorrectGeometry
+	}
+
+	p[0] = scanFloat64(data[5:13], littleEndian)
+	p[1] = scanFloat64(data[13:21], littleEndian)
+
+	return nil
+}
+
+// Scan implements the sql.Scanner interface allowing
+// line structs to be passed into rows.Scan(...interface{})
+// The column must be of type LineString and contain 2 points,
+// or an error will be returned. Data must be fetched in WKB format.
+// Will attempt to parse MySQL's SRID+WKB format if the data is of the right size.
+func (l *Line) Scan(value interface{}) error {
+	data, ok := value.([]byte)
+	if !ok {
+		return ErrUnsupportedDataType
+	}
+
+	if len(data) == 41 {
+		// the length of a 2 point linestring type in WKB
+		return l.unmarshalWKB(data)
+	}
+
+	if len(data) == 45 {
+		// Most likely MySQL's SRID+WKB format.
+		// However, could be some encoding of another type.
+		// But those would be invalid for parsing a line.
+		return l.unmarshalWKB(data[4:])
+	}
+
+	return ErrIncorrectGeometry
+}
+
+func (l *Line) unmarshalWKB(data []byte) error {
+	if len(data) != 41 {
+		return ErrNotWKB
+	}
+
+	littleEndian, typeCode, err := scanPrefix(data)
+	if err != nil {
+		return err
+	}
+
+	if typeCode != 2 {
+		return ErrIncorrectGeometry
+	}
+
+	length := scanUint32(data[5:9], littleEndian)
+	if length != 2 {
+		return ErrIncorrectGeometry
+	}
+
+	l.a[0] = scanFloat64(data[9:17], littleEndian)
+	l.a[1] = scanFloat64(data[17:25], littleEndian)
+	l.b[0] = scanFloat64(data[25:33], littleEndian)
+	l.b[1] = scanFloat64(data[33:41], littleEndian)
+
+	return nil
+}
+
+// Scan implements the sql.Scanner interface allowing
+// line structs to be passed into rows.Scan(...interface{})
+// The column must be of type LineString, Polygon or MultiPoint
+// or an error will be returned. Data must be fetched in WKB format.
+// Will attempt to parse MySQL's SRID+WKB format if obviously no WKB
+// or parsing as WKB fails.
+func (ps *PointSet) Scan(value interface{}) error {
+	data, ok := value.([]byte)
+	if !ok {
+		return ErrUnsupportedDataType
+	}
+
+	if len(data) < 6 {
+		return ErrNotWKB
+	}
+
+	// first byte of real WKB data indicates endian and should 1 or 0.
+	if data[0] == 0 || data[0] == 1 {
+		if err := ps.unmarshalWKB(data); err == nil {
+			return nil
+		}
+	}
+
+	return ps.unmarshalWKB(data[4:])
+}
+
+func (ps *PointSet) unmarshalWKB(data []byte) error {
```

Comment:
What about making an interface that PointSet, Point, and Path all implement and putting that interface in this file? Then  each geo object can implement the methods. That might make it easier for others to find the code in the library.